x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/role_mappings_table.test.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { EuiFieldSearch, EuiTableRow, EuiTextColor } from '@elastic/eui';
+
+import { wsRoleMapping, asRoleMapping } from './__mocks__/roles';
+
+import { ALL_LABEL } from './constants';
+
+import { RoleMappingsTable } from './role_mappings_table';
+
+describe('RoleMappingsTable', () => {
+  const getRoleMappingPath = jest.fn();
+  const roleMappings = [wsRoleMapping] as any;
+
+  const props = {
+    accessItemKey: 'groups' as 'groups' | 'engines',
+    accessHeader: 'access',
+    roleMappings,
+    addMappingButton: <button />,
+    shouldShowAuthProvider: true,
+    getRoleMappingPath,
+  };
+
+  it('renders', () => {
+    const wrapper = shallow(<RoleMappingsTable {...props} />);
+
+    expect(wrapper.find(EuiFieldSearch)).toHaveLength(1);
+    expect(wrapper.find(EuiTableRow)).toHaveLength(1);
+  });
+
+  it('renders auth provider display names', () => {
+    const wrapper = shallow(<RoleMappingsTable {...props} />);
+
+    expect(wrapper.find('[data-test-subj="AuthProviderDisplay"]').prop('children')).toEqual(
+      'Any, other_auth'
+    );
+  });
+
+  it('handles input change', () => {
+    const wrapper = shallow(<RoleMappingsTable {...props} />);
+    const input = wrapper.find(EuiFieldSearch);
+    const value = 'Query';
+    input.simulate('change', { target: { value } });
+
+    expect(wrapper.find(EuiTableRow)).toHaveLength(0);
+  });
+
+  it('shows default message when "accessAllEngines" is true', () => {
+    const wrapper = shallow(
+      <RoleMappingsTable {...props} roleMappings={[asRoleMapping as any]} accessItemKey="engines" />
+    );
+
+    expect(wrapper.find('[data-test-subj="AccessItemsList"]').prop('children')).toEqual(ALL_LABEL);
+  });
+
+  it('handles display when no items present', () => {
+    const noItemsRoleMapping = { ...asRoleMapping };
+    noItemsRoleMapping.accessAllEngines = false;
+
+    const wrapper = shallow(
+      <RoleMappingsTable
+        {...props}
+        roleMappings={[noItemsRoleMapping as any]}
+        accessItemKey="engines"
+      />
+    );
+
+    expect(wrapper.find('[data-test-subj="AccessItemsList"]').prop('children')).toEqual(
```

Comment:
I don't know if this is better or worse ... but i you want to access the children as a wrapper and make a text assertion on just the `-` text, you could do something like:

```
expect(wrapper.find('[data-test-subj="AccessItemsList"]').children().children().text()).toEqual('-')
```